h2o-algos/src/main/java/hex/tree/SharedTree.java

```diff
@@ -155,7 +155,7 @@ protected void ignoreInvalidColumns(int npredictors, boolean expensive) {
     if( !(0. < _parms._col_sample_rate_change_per_level && _parms._col_sample_rate_change_per_level <= 2) )
       error("_col_sample_rate_change_per_level", "col_sample_rate_change_per_level must be > 0" +
               " and <= 2");
-    if (_train != null) {
+    if (_train != null && _train.find(_parms._weights_column) != -1) {
```

Comment:
should we not change the implementation of `hasWeightCol()` instead?
Well, maybe not… because once validation is done, we shouldn't need to do this `find` that can be inefficient on wide datasets.

Maybe the condition should be merged with the call to `hasWeightCol()` below though: if `weight_column=null` now, then `sumWeights=nrows` and we don't validate anymore that `nrows` must be `>= 2 * min_rows`.
wdyt?

Reply:
The validation of the `_weights_column` is in `ModelBuilder.init()`,  but after calling `error()` it continues to process the whole `init()` method, and after that, it catches the Exception. 

https://github.com/h2oai/h2o-3/blob/7992e88994590f6775ef8f543d52ad64bc3f6d84/h2o-core/src/main/java/hex/ModelBuilder.java#L1081-L1101

But yes, you are correct, we need to merge the condition below, not to skip `_min_rows` check due to missing `_weights_column`. :+1:

Reply:
@sebhrusen, WDYT about this: https://github.com/h2oai/h2o-3/pull/5630/commits/2ddc61178895b10fa37c5e0190909f8c6e22ba44?

The `_weights` should be defined from `ModelBuilder.init()`, if it is not initialized and `hasWeightCol()` is true, something is wrong.

Reply:
even better yes! looks like the call to `hasWeightsCol()` is not even necessary anymore.

Reply:
Fixed.